### PR TITLE
Add sasl support for logging kafka

### PR DESCRIPTION
**Pull request by aiwantaozi**
Add sasl support for logging kafka

Problem:
need support sasl auth for kafka logging

Solution:
add sasl support

Issue:
https://github.com/rancher/rancher/issues/16614

Types:
https://github.com/rancher/types/pull/683

### CODE REVIEW

File: pkg/api/server/managementstored/setup.go [hunk 1]
```diff
@@ -594,9 +596,9 @@ func MultiClusterApps(schemas *types.Schemas, management *config.ScaledContext)
 		MultiClusterApps:              management.Management.MultiClusterApps(""),
 		MultiClusterAppLister:         management.Management.MultiClusterApps("").Controller().Lister(),
 		MultiClusterAppRevisionLister: management.Management.MultiClusterAppRevisions("").Controller().Lister(),
-		Users:                         management.Management.Users(""),
-		PrtbLister:                    management.Management.ProjectRoleTemplateBindings("").Controller().Lister(),
-		RoleTemplateLister:            management.Management.RoleTemplates("").Controller().Lister(),
+		Users:              management.Management.Users(""),
+		PrtbLister:         management.Management.ProjectRoleTemplateBindings("").Controller().Lister(),
+		RoleTemplateLister: management.Management.RoleTemplates("").Controller().Lister(),
```
Comment (fyery-chen) [inline pkg/api/server/managementstored/setup.go]: Nit: go fmt

File: pkg/controllers/user/logging/generator/cluster_template.go [hunk 2]
```diff
@@ -179,6 +179,7 @@ var ClusterTemplate = `{{ if .clusterTarget.CurrentTarget }}
     output_data_type  "json"
     output_include_tag true
     output_include_time true
+    max_send_retries  3
```
[the remark this answers is not in the document]
Comment (aiwantaozi) [reply pkg/controllers/user/logging/generator/cluster_template.go]: no, I just move line from below to current line

File: vendor/github.com/rancher/types/apis/management.cattle.io/v3/globaldns_types.go [hunk 3]
```diff
@@ -40,12 +40,19 @@ type GlobalDNSProvider struct {
...
 	RootDomain string `json:"rootDomain" norman:"required"`
 	AccessKey  string `json:"accessKey"`
 	SecretKey  string `json:"secretKey" norman:"type=password"`
 }
+
+type CloudflareProviderConfig struct {
+	RootDomain string `json:"rootDomain" norman:"required"`
+	APIKey     string `json:"apiKey" norman:"required,type=password"`
+	APIEmail   string `json:"apiEmail" norman:"required"`
+}
```
[the remark this answers is not in the document]
Comment (aiwantaozi) [reply vendor/github.com/rancher/types/apis/management.cattle.io/v3/globaldns_types.go]: Just because update the types to latest one.

**Comment by fyery-chen** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Comment by cjellick** [comment]
@aiwantaozi @fyery-chen @loganhz Is this going to require a UI change? Seems like it should. If so, you shoud open a corresponding UI issue with the label `area/ui`.

**Comment by cjellick** [comment]
Closing in favor of https://github.com/rancher/rancher/pull/17542, which is the same except that I updated vendor.conf appropriately

**Pull request closed without merging**